File: python_projects/Advanced_basic/password_strength_checker.py
import string

def check_strength(password):
    strength = 0
    if len(password) >= 8:
        strength += 1
    if any(c.isdigit() for c in password):
        strength += 1
    if any(c in string.ascii_uppercase for c in password):
        strength += 1
    if any(c in string.punctuation for c in password):
        strength += 1

    print("\n🔍 Password analysis:")
    if strength == 4:
        print("🟢 Strong Password")
    elif strength == 3:
        print("🟡 Moderate Password")
    else:
        print("🔴 Weak Password")

password = input("Enter a password to check: ")
check_strength(password)

File: java_projects/Advanced_basic/PasswordStrengthChecker.java
import java.util.Scanner;

public class PasswordStrengthChecker {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a password: ");
        String password = sc.nextLine();

        int strength = 0;
        if (password.length() >= 8) strength++;
        if (password.matches(".*\\d.*")) strength++;
        if (password.matches(".*[A-Z].*")) strength++;
        if (password.matches(".*[!@#$%^&*()].*")) strength++;

        System.out.print("🔍 Strength: ");
        switch (strength) {
            case 4 -> System.out.println("🟢 Strong Password");
            case 3 -> System.out.println("🟡 Moderate Password");
            default -> System.out.println("🔴 Weak Password");
        }

        sc.close();
    }
}
